Argo/Functions/decode.swift

```diff
@@ -1,35 +1,215 @@
-// MARK: decode root objects and arrays
+/**
+  Decode a raw JSON value into an optional `Decodable` value
 
+  This function takes the output from `NSJSONSerialization` and attempts to
+  decode it into a `Decodable` value, returning an `Optional`. This works based
+  on the type you ask for.
+
+  For example, the following code attempts to decode an optional `String`,
+  because that's what we have explicitly stated is the return type:
+
+  ```
+  do {
+    let object = try NSJSONSerialization.JSONObjectWithData(data, options: nil)
+    let str: String? = decode(object)
+  } catch {
+    // handle error
+  }
+  ```
+
+  - parameter object: The output from `NSJSONSerialization`
+  - returns: An `Optional` value of type `T` where `T` is `Decodable`
+*/
 public func decode<T: Decodable where T == T.DecodedType>(object: AnyObject) -> T? {
   return decode(object).value
 }
 
+/**
+  Decode a raw JSON value into an optional array of `Decodable` values
```

Comment:
This is probably a bigger discussion about our language for the project. I _still_ don't think we've come up with a agreed upon word for the `<|` family of operators. I have been using `decode` there, but it _is_ a really overloaded term in this lib. I'm also not sure that I totally understand what the difference is between decoding and parsing.

I also think it'd be weird to say that the `decode` function "Parses a raw JSON value into an optional array of `Decodable` values"